src/renderer/components/TaskProgress.ts

```diff
@@ -0,0 +1,85 @@
+import { Task, TaskRuntime, TaskHandle } from "@xmcl/task";
+
+import taskProgressTemplate from "../templates/TaskProgress.pug";
+import { ResolvedVersion } from "@xmcl/core";
+
+export default class TaskProgress extends HTMLDivElement {
+	private $progress = () => this.getElementsByClassName("ui progress")[0];
+
+	public constructor() {
+		super();
+	}
+
+	public connectedCallback(): void {
+		if (!this.hasChildNodes())
+			this.render();
+	}
+
+	public render(): void {
+		this.innerHTML = taskProgressTemplate();
+		$(this.$progress()).progress();
+	}
+
+	public addInstallTask(task: Task<ResolvedVersion>, instanceName: string): TaskRuntime<Task.State> {
+		const runtime = Task.createRuntime();
+		let rootTask: Task.State;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['rootTask' is assigned a value but never used.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=5247069)